openedx/core/djangoapps/content_libraries/api.py

```diff
@@ -1952,3 +1952,28 @@ def import_blocks_create_task(library_key, course_key, use_course_key_as_block_i
     log.info(f"Import block task created: import_task={import_task} "
              f"celery_task={result.id}")
     return import_task
+
+
+def create_or_update_xblock_upstream_link(xblock, course_key: str, course_name: str, created: datetime | None = None):
+    """
+    Create or update upstream->downstream link in database for given xblock.
+    """
+    if not xblock.upstream:
+        log.info(f"No upstream found for xblock: {xblock.usage_key}")
```

Comment:
nit: this will generate a lot of log messages when people change course names, because most course blocks aren't upstream-linked.
```suggestion
```